iot_packet_verifier/src/burner.rs

```diff
@@ -125,18 +125,18 @@ where
             .remove_pending_transaction(txn.get_signature())
             .await?;
         pending_tables_txn
-            .subtract_burned_amount(&payer, amount)
+            .subtract_burned_amount(&escrow_key, amount)
             .await?;
         pending_tables_txn.commit().await?;
 
         let mut balance_lock = self.balances.lock().await;
-        let payer_account = balance_lock.get_mut(&payer).unwrap();
+        let payer_account = balance_lock.get_mut(&escrow_key).unwrap();
         // Reduce the pending burn amount and the payer's balance by the amount
         // we've burned.
         payer_account.burned = payer_account.burned.saturating_sub(amount);
         payer_account.balance = payer_account.balance.saturating_sub(amount);
 
-        metrics::counter!("burned", "payer" => payer.to_string()).increment(amount);
+        metrics::counter!("burned", "payer" => escrow_key.to_string()).increment(amount);
```

Comment:
```suggestion
        metrics::counter!("burned", "payer" => %escrow_key).increment(amount);
```
this should already be a string now, no?